python/lib/dependabot/python/file_parser.rb

```diff
@@ -294,10 +294,39 @@ def group_from_filename(filename)
 
       def blocking_marker?(dep)
         return false if dep["markers"] == "None"
-        return true if dep["markers"].include?("<")
-        return false if dep["markers"].include?(">")
 
-        dep["requirement"]&.include?("<")
+        marker = dep["markers"]
+        version = python_raw_version
+
+        if marker.include?("python_version")
+          !marker_satisfied?(marker, version)
+        else
+          return true if dep["markers"].include?("<")
+          return false if dep["markers"].include?(">")
+
+          dep["requirement"]&.include?("<")
+        end
+      end
+
+      def marker_satisfied?(marker, python_version)
+        return true if marker == "None"
+
+        operator, version = marker.match(/([<>=!]=?)\s*"?([\d.]+)"?/).captures
+
+        case operator
+        when "<"
+          Gem::Version.new(python_version) < Gem::Version.new(version)
```

Comment:
Please don't use Gem::Version for this. Use Dependabot::Python::Version instead. You could run into some strange and nasty issues with these comparisons due to the Gem::Version spec being different from the Dependabot::Python::Version  spec.